Upcoming movies

gustavo-villar-globant opened the pull request.

gustavo-villar-globant committed:
Added upcoming movies web service call

gustavo-villar-globant committed:
Merge branch 'master' into upcoming_movies

gustavo-villar-globant committed:
Added MoviesList

gustavo-villar-globant committed:
Added Home scene as parent from Now Playing and Upcoming Lists

gustavo-villar-globant committed:
changed background color for images

gustavo-villar-globant committed:
fixed movie list transition

gustavo-villar-globant committed:
Moved segmented control on Movies Home to the navigation bar

gustavo-villar-globant committed:
Added Interactive Transition to MoviesHomeViewController

chmoncada commented on Cinebase/Core/Formatters/DateFormatter.swift:
```diff
@@ -0,0 +1,32 @@
+    private let formatter = Foundation.DateFormatter()
+    
+    func format(_ date: Date, style: Style) -> String {
+        switch style {
```
is there a way to put the values of each style as individual vars, it will be cleaner to modify?

gustavo-villar-globant replied:
What do you mean by values? the date format?

chmoncada replied:
yes 
var dateStyle
var timeStyle

each one with the switch, what do you think? i think it will be better to add and modify new formatter styles, what do you think?

chmoncada commented on Cinebase/Gateways/WebAPI/Parsing/MovieParser.swift:
```diff
@@ -20,19 +20,23 @@ class MovieParser: ModelParser {
+            let releaseDate = dateParser.parse(from: releaseDateString) else {
                 let parsingError = ParsingError<Movie>(json: json)
                 return .failure(parsingError)
         }
```
It will be better to put a default value as ND if there is a error in data parsing, it can be a case where there is a knowledge of future movie but not a specific date of release

gustavo-villar-globant replied:
What would you use as a default value? It can't be the current date

gustavo-villar-globant replied:
Maybe I could make it an optional and give it a default value of nil

chmoncada replied:
default value should be "NA" Not available or "TBD" To be determined

gustavo-villar-globant committed:
Fixed Tests
Changed movie release date to be optional
Improved implementation of DateFormatter

gustavo-villar-globant committed:
Changed text to english
